File: QuanLyQuanAnPython/views/thong_ke_panel.py
import tkinter as tk
from tkinter import ttk, messagebox
from datetime import datetime
from service import ThongKeService
import utils

class ThongKePanel(tk.Frame):
    def __init__(self):
        super().__init__()
        self.init_ui()
        self.load_thong_ke(datetime.now())
    
    def init_ui(self):
        # Panel trên cùng để chọn ngày
        top_frame = tk.Frame(self)
        top_frame.pack(fill=tk.X, padx=10, pady=10)
        
        tk.Label(top_frame, text="Ngày:").pack(side=tk.LEFT, padx=5)
        
        # Date entry
        self.ngay_entry = tk.Entry(top_frame, width=15)
        self.ngay_entry.pack(side=tk.LEFT, padx=5)
        self.ngay_entry.insert(0, datetime.now().strftime("%d/%m/%Y"))
        
        btn_xem = tk.Button(top_frame, text="Xem", command=self.xem_thong_ke)
        btn_xem.pack(side=tk.LEFT, padx=5)
        
        # Panel thống kê chính
        main_frame = tk.Frame(self)
        main_frame.pack(fill=tk.BOTH, expand=True, padx=10, pady=10)
        
        # Panel trái - thông số
        info_frame = tk.LabelFrame(main_frame, text="THỐNG KÊ", font=("Arial", 12, "bold"))
        info_frame.pack(side=tk.LEFT, fill=tk.BOTH, expand=True, padx=5)
        
        # Doanh thu
        tk.Label(info_frame, text="Doanh thu:", font=("Arial", 12)).grid(row=0, column=0, padx=10, pady=10, sticky="w")
        self.lb_doanh_thu = tk.Label(info_frame, text="0 ₫", font=("Arial", 14, "bold"), fg="red")
        self.lb_doanh_thu.grid(row=0, column=1, padx=10, pady=10)
        
        # Số hóa đơn
        tk.Label(info_frame, text="Số hóa đơn:", font=("Arial", 12)).grid(row=1, column=0, padx=10, pady=10, sticky="w")
        self.lb_so_hoa_don = tk.Label(info_frame, text="0", font=("Arial", 14))
        self.lb_so_hoa_don.grid(row=1, column=1, padx=10, pady=10)
        
        # Tổng món bán
        tk.Label(info_frame, text="Tổng món bán:", font=("Arial", 12)).grid(row=2, column=0, padx=10, pady=10, sticky="w")
        self.lb_so_mon = tk.Label(info_frame, text="0", font=("Arial", 14))
        self.lb_so_mon.grid(row=2, column=1, padx=10, pady=10)
        
        # Panel phải - top món
        top_mon_frame = tk.LabelFrame(main_frame, text="TOP 3 MÓN BÁN CHẠY", font=("Arial", 12, "bold"))
        top_mon_frame.pack(side=tk.RIGHT, fill=tk.BOTH, expand=True, padx=5)
        
        self.list_top_mon = tk.Listbox(top_mon_frame, font=("Arial", 12), height=10)
        self.list_top_mon.pack(fill=tk.BOTH, expand=True, padx=10, pady=10)
    
    def xem_thong_ke(self):
        try:
            ngay_str = self.ngay_entry.get()
            ngay = datetime.strptime(ngay_str, "%d/%m/%Y")
            self.load_thong_ke(ngay)
        except:
            messagebox.showerror("Lỗi", "Vui lòng nhập đúng định dạng dd/mm/yyyy")
    
    def load_thong_ke(self, ngay):
        doanh_thu = ThongKeService.get_doanh_thu(ngay)
        so_hoa_don = ThongKeService.get_so_hoa_don(ngay)
        so_mon = ThongKeService.get_tong_mon_ban(ngay)
        top_mon = ThongKeService.get_top_mon_ban_chay(ngay, 3)
        
        self.lb_doanh_thu.config(text=utils.format_vnd(doanh_thu))
        self.lb_so_hoa_don.config(text=str(so_hoa_don))
        self.lb_so_mon.config(text=str(so_mon))
        
        self.list_top_mon.delete(0, tk.END)
        for i, (ten, so_luong) in enumerate(top_mon, 1):
            self.list_top_mon.insert(tk.END, f"{i}. {ten} - {so_luông} phần")

File: DACS1/src/view/ThongKePanel.java
package view;

import service.ThongKeService;
import utils.DateUtils;
import utils.NumberUtils;

import javax.swing.*;
import java.awt.*;
import java.util.Date;
import java.util.List;
import java.util.Map;

public class ThongKePanel extends JPanel {
    private JLabel lblDoanhThu, lblSoHoaDon, lblSoMon;
    private JList<String> listTopMon;
    private DefaultListModel<String> listModel;
    private JSpinner spnNgay;
    
    public ThongKePanel() {
        initUI();
        loadThongKe(new Date());
    }
    
    private void initUI() {
        setLayout(new BorderLayout());
        setBorder(BorderFactory.createEmptyBorder(10, 10, 10, 10));
        
        // Panel chọn ngày
        JPanel topPanel = new JPanel(new FlowLayout(FlowLayout.LEFT));
        topPanel.add(new JLabel("Chọn ngày:"));
        
        SpinnerDateModel dateModel = new SpinnerDateModel();
        spnNgay = new JSpinner(dateModel);
        JSpinner.DateEditor dateEditor = new JSpinner.DateEditor(spnNgay, "dd/MM/yyyy");
        spnNgay.setEditor(dateEditor);
        spnNgay.setValue(new Date());
        topPanel.add(spnNgay);
        
        JButton btnXem = new JButton("Xem thống kê");
        btnXem.addActionListener(e -> {
            Date ngay = (Date) spnNgay.getValue();
            loadThongKe(ngay);
        });
        topPanel.add(btnXem);
        
        add(topPanel, BorderLayout.NORTH);
        
        // Panel chính
        JPanel mainPanel = new JPanel(new GridLayout(1, 2, 10, 0));
        
        // Panel thông số
        JPanel infoPanel = new JPanel(new GridBagLayout());
        infoPanel.setBorder(BorderFactory.createTitledBorder("THỐNG KÊ"));
        GridBagConstraints gbc = new GridBagConstraints();
        gbc.insets = new Insets(15, 15, 15, 15);
        gbc.fill = GridBagConstraints.HORIZONTAL;
        
        gbc.gridx = 0; gbc.gridy = 0;
        infoPanel.add(new JLabel("Doanh thu:"), gbc);
        lblDoanhThu = new JLabel("0 ₫");
        lblDoanhThu.setFont(new Font("Arial", Font.BOLD, 18));
        lblDoanhThu.setForeground(Color.RED);
        gbc.gridx = 1;
        infoPanel.add(lblDoanhThu, gbc);
        
        gbc.gridx = 0; gbc.gridy = 1;
        infoPanel.add(new JLabel("Số hóa đơn:"), gbc);
        lblSoHoaDon = new JLabel("0");
        lblSoHoaDon.setFont(new Font("Arial", Font.BOLD, 16));
        gbc.gridx = 1;
        infoPanel.add(lblSoHoaDon, gbc);
        
        gbc.gridx = 0; gbc.gridy = 2;
        infoPanel.add(new JLabel("Tổng món bán:"), gbc);
        lblSoMon = new JLabel("0");
        lblSoMon.setFont(new Font("Arial", Font.BOLD, 16));
        gbc.gridx = 1;
        infoPanel.add(lblSoMon, gbc);
        
        // Panel top món
        JPanel topMonPanel = new JPanel(new BorderLayout());
        topMonPanel.setBorder(BorderFactory.createTitledBorder("TOP 3 MÓN BÁN CHẠY"));
        listModel = new DefaultListModel<>();
        listTopMon = new JList<>(listModel);
        listTopMon.setFont(new Font("Arial", Font.PLAIN, 14));
        listTopMon.setFixedCellHeight(40);
        topMonPanel.add(new JScrollPane(listTopMon), BorderLayout.CENTER);
        
        mainPanel.add(infoPanel);
        mainPanel.add(topMonPanel);
        
        add(mainPanel, BorderLayout.CENTER);
    }
    
    private void loadThongKe(Date ngay) {
        double doanhThu = ThongKeService.getDoanhThu(ngay);
        int soHoaDon = ThongKeService.getSoHoaDon(ngay);
        int soMon = ThongKeService.getTongMonBan(ngay);
        
        lblDoanhThu.setText(NumberUtils.formatVND(doanhThu));
        lblSoHoaDon.setText(String.valueOf(soHoaDon));
        lblSoMon.setText(String.valueOf(soMon));
        
        listModel.clear();
        List<Map.Entry<String, Integer>> topMon = ThongKeService.getTopMonBanChay(ngay, 3);
        int i = 1;
        for (Map.Entry<String, Integer> entry : topMon) {
            listModel.addElement(i + ". " + entry.getKey() + " - " + entry.getValue() + " phần");
            i++;
        }
        if (topMon.isEmpty()) {
            listModel.addElement("Chưa có dữ liệu");
        }
    }
}